main/src/main/scala/sbt/Defaults.scala

```diff
@@ -3786,7 +3786,7 @@ object Classpaths {
 
   private[sbt] def isScala2Scala3Sandwich(sbv1: String, sbv2: String): Boolean = {
     def compare(a: String, b: String): Boolean =
-      a == "2.13" && (b.startsWith("0.") || b.startsWith("3.0"))
+      a == "2.13" && (b.startsWith("0.") || b.startsWith("3"))
```

Comment:
I thought we don't have concrete guarantee of 2.13 compatibility beyond `3.0`.

Reply:
We have concrete guarantee until 3.T for some unknown T, but at that point we'll bump Tasty major too, so incompatible projects will fail at compile-time.

Reply:
... and also this is about the scalaBinaryVersion, which we'll bump when we go to 3.T.
Until we break our own binary compatibility, we'll keep supporting 2.13.

Reply:
Does it work both ways?
We can make 3.x compatible with _2.13 but we will need a 2.13.y to consume each 3.x.

Reply:
In that direction the compiler will check the tasty version to determine if it's compatible or not

Reply:
ok sounds good.